### PR TITLE
[Misc][Simulator] Update vllm simulator backend

### PR DESCRIPTION
**Pull request by ZeldaHuang**
1.  record simulator profiling data in `InstanceInfo`
2. update `get_instance_info` to get correct info
3. update `BackendSimVLLM`
4. add simulator docs.

### CODE REVIEW

File: llumnix/backends/profiling.py [hunk 1]
```diff
@@ -205,29 +202,25 @@ def get_latency_mem(backend_type: BackendType, profiling_database: ProfilingData
...
+    parser.add_argument("--log-csv-path", type=str, required=True)
+    parser.add_argument("--model", type=str, help="filename of your model, like 'Meta-Llama-3-8B-Instruct'")
     parser.add_argument("--gpu", type=str, default="a10")
     parser.add_argument("--tp", type=int, default=1)
     parser.add_argument("--pp", type=int, default=1)
     parser.add_argument("--gpu-memory-utilization", type=float, default=0.9)
     parser.add_argument("--block-size", type=int, default=16)
     parser.add_argument("--max-num-batched-tokens", type=int, default=8000)
-    parser.add_argument("--num-gpu-blocks", type=int, default=0)
+    parser.add_argument("--num-gpu-blocks", type=int, required=True, help="kv cache blocks number")
```
Comment (s5u13b) [inline llumnix/backends/profiling.py:213]: if --num-gpu-blocks is required, why --gpu is not required?

File: llumnix/backends/vllm/scheduler.py [hunk 2]
```diff
@@ -21,8 +21,9 @@
 
 from llumnix.instance_info import InstanceInfo
 from llumnix.logger import init_logger
-from llumnix.llumlet.request import LlumnixRequest, RequestInferenceType
```
Comment (s5u13b) [inline llumnix/backends/vllm/scheduler.py:24]: so LlumnixRequest is only used in llumlet layer now?

**Review by s5u13b — APPROVED** [review]

File: llumnix/backends/profiling.py (hunk 1 above)
Comment (ZeldaHuang) [reply llumnix/backends/profiling.py:213]: Maybe remove --gpu is better? It's not a engine related args.

File: llumnix/backends/vllm/scheduler.py (hunk 2 above)
Comment (ZeldaHuang) [reply llumnix/backends/vllm/scheduler.py:24]: yes

File: tests/unit_test/backends/vllm/test_simulator.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                blocks_to_swap_out=out.blocks_to_swap_out,
+                blocks_to_copy=out.blocks_to_copy,
+                num_lookahead_slots=out.num_lookahead_slots,
+                running_queue_size=out.running_queue_size,
+            )
+    outputs = executor.execute_model(execute_model_req)
+    assert len(outputs[0].outputs) == 2
+
+def test_backend():
+    pass
```
Comment (KuilongCui) [inline tests/unit_test/backends/vllm/test_simulator.py:49]: adding a TODO maybe better

**Review by KuilongCui — APPROVED** [review]

**Pull request merged**